client/app/main/student/student.controller.js

```diff
@@ -109,6 +109,50 @@ function StudentCtrl($scope, $stateParams, Intervention, Modal, Outreach,
       toastr.error(err);
     });
   };
+  $scope.toggleArchiveOutreach = function(outreach) {
+    var oldValue = outreach.archived;
+    var promise = Outreach.toggleArchive({
+      id: outreach._id
+    }, {
+      archived: !(outreach.archived)
+    }).$promise;
+    promise.then(function(outreach) {
+      $scope.$evalAsync(function() {
+        var outreaches = $scope.student.outreaches
+        outreaches.map(function(oldOutreach, i) {
+            if(oldOutreach._id === outreach._id) {
+              outreaches[i].archived = outreach.archived;
+            }
+        });
+      });
+      toastr.info(
+        'The ' + outreach.type +
+        ' outreach has been' +
+        (outreach.archived === true ? ' archived.' : ' unarchived.'));
+    }, function(err) {
+      outreach.archived = oldValue;
+      toastr.error(err);
+    });
+  };
+  $scope.deleteOutreach = function(outreach) {
+    var model = outreach;
+    var deleteFn = function(model) {
+      Outreach.delete({id: model._id});
+      var outreaches = $scope.student.outreaches
+      outreaches.map(function(oldOutreach, i) {
```

Comment:
`pullAt` will pull at a specific index. Since you don't know what index that outreach is using `pull` with match on the `_id` is probably the way to go. The other option is to pass in the index to the surrounding function. I think just using pull is probably the way to go.